### PR TITLE
#82 imperial statistics

### PR DESCRIPTION
**Pull request by richardcrete**

### CODE REVIEW

**Comment by florian987** [comment]
thanks!

what do you think on doing the transformation at the view level, using twig globals and functions?

**Comment by richardcrete** [comment]
That's a good idea, I'll do the changes!

File: src/Controller/ParkController.php [hunk 1]
```diff
@@ -5,16 +5,18 @@
...
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 
 class ParkController extends AbstractController
 {
     /** Show park details. */
     #[Route(path: '/parks/{slug}', name: 'park_show', options: ['expose' => true], methods: ['GET'])]
-    public function showAction(ParkRepository $parkRepository, Park $park): Response
+    public function showAction(ParkRepository $parkRepository, Park $park, Request $request, CoasterRepository $coasterRepository): Response
```
Comment (florian987) [inline src/Controller/ParkController.php:19]: not used

File: src/Controller/CoasterController.php [hunk 2]
```diff
@@ -111,25 +104,21 @@ public function ajaxLoadImages(EntityManagerInterface $em, Coaster $coaster, int
...
     public function showAction(
-        Request                 $request,
-        Coaster                 $coaster,
-        RiddenCoasterRepository $riddenCoasterRepository
-    ): Response
-    {
+        Request $request,
+        Coaster $coaster,
+        RiddenCoasterRepository $riddenCoasterRepository,
+        CoasterRepository $coasterRepository,
```
Comment (florian987) [inline src/Controller/CoasterController.php:120]: to be removed

File: config/packages/twig.yaml [hunk 3]
```diff
@@ -3,6 +3,7 @@ twig:
     globals:
         locales: '%app_locales_array%'
         pictures_cdn: '%env(PICTURES_CDN)%'
+        unities: '@App\Twig\UnitiesExtension'
```
Comment (florian987) [inline config/packages/twig.yaml:6]: units?

File: templates/Coaster/show.html.twig [hunk 4]
```diff
@@ -93,15 +93,23 @@
             <button type="button" class="btn btn-default btn-block btn-float btn-float-lg text-bold">
               <i class="icon-arrow-up8 text-warning-600"></i>
               {% if coaster.height is not empty %}
-                {{ coaster.height ~ ' m' }}
+                  {% if unities.isImperial(app.request.locale) %}
+                      {{ (coaster.height * 3.281)|round(0) }} ft
```
Comment (florian987) [inline templates/Coaster/show.html.twig:97]: why not a format_unit function, doing the conversion centrally, and adding the extension also?

**Review by florian987 — CHANGES_REQUESTED** [review]
thank you, i suggested other changes :)

**Pull request closed without merging**